tests/testutility.nim

```diff
@@ -71,3 +72,75 @@ suite "Utility":
   test "unsuccessful safeConvert from uint to int":
     check not (compiles do:
       result: uint = safeConvert[int, uint](11.uint))
+
+  test "withValue":
```

Comment:
This test would benefit from being split into smaller tests with descriptive names that test only one thing like "withValue calls right branch when Opt is some", "withValue calls right branch when Opt is none", "withValue is called only once when Opt is some", etc.

Reply:
Done :+1: